backend/route/views.py

```diff
@@ -1,3 +1,269 @@
-#from django.shortcuts import render
+from django.http import HttpResponse, HttpResponseNotAllowed, JsonResponse,HttpResponseBadRequest
+from django.contrib.auth.models import User
+from django.views.decorators.csrf import ensure_csrf_cookie
+import json
+from .models import *
+from json.decoder import JSONDecodeError
 
-# Create your views here.
+def posts(request):
+    if request.method!='GET' and request.method!='POST':
+        return HttpResponseNotAllowed(['GET', 'POST'])
+    elif request.method=="GET":
+        postlist=[]
+        for post in Post.objects.all():
+            comments=[]
+            for comment in post.comment_set.all():
+                comments.append({'content': comment.content, 'author_id':comment.author.id})
+            folder_id=''
+            if post.folder:
+                folder_id=post.folder.id
+            postlist.append({'title': post.title, 'author_id': post.author.id, 'header_img': post.header_image.url, 'thumbnail_image': post.thumbnail_image.url, 
+            'days': post.days, 'folder_id': folder_id, 'is_shared':post.is_shared, 'theme':post.theme, 'comment': comments, 'season': post.season, 
+            'location': post.location, 'availableWithOutCar': post.availableWithoutCar})
+        if len(postlist)==0:
+            return HttpResponse(status=404)
+        return JsonResponse(postlist, safe=False)
+    else: #POST
+        logged_user_id=request.session.get('user', None)
+        if not logged_user_id:
+            return HttpResponse(status=405)
+        try:
+            body = request.body.decode()
```

Comment:
How about initializing `body` value as `json.loads(request.body.decode())`?

I think it is more efficient in that `json.loads` method executed only once.